### PR TITLE
Ahmet homework3

### PR DESCRIPTION
**Pull request by ahmthsrvsmz**

### CODE REVIEW

**Comment by ahmthsrvsmz** [comment]
This is my last homework. I copied from lecture file and I just created 4 more actions inside ~/src/index.js and tried on the postman. It is all work. 👍

File: week3/homework/src/index.js [hunk 1]
```diff
@@ -1,3 +1,88 @@
...
+const TODO_SLUG = 'todos';
+
+const todo = new Todo(FILENAME);
+
+const app = new Express();
+
+// Use built-in JSON middleware to automatically parse JSON
+app.use(Express.json());
+
+// Functions that I created
```
Comment (gajduk) [inline week3/homework/src/index.js]: thanks for the note

File: week3/homework/src/index.js [hunk 2]
```diff
@@ -1,3 +1,88 @@
...
+function readTodo(todo, request, response) {
+  todo.read().then(todos => {
+    const todoId = request.params.id;
+    const pickTodo = todos.find(todo => todo.id === todoId);
```
Comment (gajduk) [inline week3/homework/src/index.js]: nice, finally someone who does not abuse map or filter to find a single element 🥇

File: week3/homework/src/index.js [hunk 3]
```diff
@@ -1,3 +1,88 @@
...
+    if (pickTodo) {
+      response.send(pickTodo);
+    } else {
+      response.status(404).send({ message: `user with id ${todoId} not found` });
```
Comment (gajduk) [inline week3/homework/src/index.js]: 👍 correct status code and clear message

File: week3/homework/src/index.js [hunk 4]
```diff
@@ -1,3 +1,88 @@
...
+      response.status(500);
+      response.json({ error: message });
+    });
+}
+
+function markAsDone(todo, req, res) {
+  const { id } = req.params;
+  const done = true;
+  todo
+    .updateDone(id, done)
```
Comment (gajduk) [inline week3/homework/src/index.js]: nice that you are reusing this function

File: week3/homework/src/index.js [hunk 5]
```diff
@@ -1,3 +1,88 @@
...
+function markAsNotDone(todo, req, res) {
+  const { id } = req.params;
+  const done = false;
+  todo
+    .updateDone(id, done)
+    .then(todo => {
+      res.send(todo);
+    })
+    .catch(error => {
+      res.status(404);
```
Comment (gajduk) [inline week3/homework/src/index.js]: is this status code always correct? What if for example a file operation fails for some reason?

File: week3/homework/src/index.js [hunk 6]
```diff
@@ -1,3 +1,88 @@
...
+      res.send({ error: error.message });
+    });
+}
+
+app.post(`/${TODO_SLUG}`, createTodo.bind(null, todo));
```
Comment (gajduk) [inline week3/homework/src/index.js]: why don't you use `todo.createTodo` here? It is shorter than using `bind`.

**Pull request closed without merging**